File: 2125.银行中的激光束数量.py
#
# @lc app=leetcode.cn id=2125 lang=python3
#
# [2125] 银行中的激光束数量
#

# @lc code=start
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        res = 0
        tmp = []
        for i in range(len(bank)):
            if j := bank[i].count('1'):
                tmp.append(j)
        if len(tmp) == 1:
            return 0
        for i in range(len(tmp) - 1):
            res += tmp[i] * (tmp[i] - 1)
        return res 


# @lc code=end

File: 2125.银行中的激光束数量.go
/*
 * @lc app=leetcode.cn id=2125 lang=golang
 *
 * [2125] 银行中的激光束数量
 */

// @lc code=start
func numberOfBeams(bank []string) int {
	res := 0
	tmp := []int{}
	for i := 0; i < len(bank); i++ {
		j := strings.Count(bank[i], "1")
		if j > 0 {
			tmp = append(tmp, j)
		}
	}
	for i := 0; i < len(tmp)-1; i++ {
		res += tmp[i] * tmp[i+1]
	}
	return res

}

// @lc code=end

